### ISSUE TITLE
localhost entries missing on OS X

### ISSUE THREAD

**Issue by lucianf** [StevenBlack/hosts#86]
On OS X, the localhost definition at the top of /etc/hosts is a bit more verbose:

```
127.0.0.1       localhost
255.255.255.255 broadcasthost
::1             localhost
fe80::1%lo0     localhost
```
In the generated hosts file, the broadcasthost (see [this](https://macosx.com/threads/broadcasthost.245946/)) and the second IPv6 are missing:

```
127.0.0.1 localhost
::1 localhost
```

**Comment by StevenBlack** [comment]
Is this really a problem, do you think?

**Comment by lucianf** [comment]
See my comment on #87, I believe the correct approach is to leave the default OS config untouched, and only expand it further.  If anything is to be changed leave it to the local admin; a tool shouldn't be opinionated and change system configuration beyond its scope (which in this case is malware/ad blocking, and those entries above have zero to do with that).

**Comment by StevenBlack** [comment]
OK.

So how can we tell which hosts the OS comes with?

Presently we backup the pre-existing hosts file, and there is provision for including additional hosts within a file named `myhosts`.

**Comment by lucianf** [comment]
I see.  The problem is that you're not assuming that `myhosts` can contain the header localhost definitions, so you're rather hardcoding it in the tool.  Not sure this is right, since different OS's will have different things in the header.  On OS X I have what I listed in my original comment, but other people on other architectures will most likely have different stuff there.  I for one would not hardcode anything, but rather expect people to put their entire `/etc/hosts` into `myhosts`.  That way you would safeguard those entries against any potential changes coming from the `sources`.

Sorry if this may sound like an academic discussion by now.

**Comment by StevenBlack** [comment]
Nah, this is a great conversation to have.  And I agree with you 100% on all of this.

I *think* all the angles are covered but it requires the end-user to know about `myhosts` if they generate the amalgamated file themselves.  But for users that download and use the hosts file as-is, that's different.

I'm thinking now that the public downloadable amalgamated hosts file should contain all the host records from all OS that we can find, just so it plays nicely everywhere.

`updateHostsFile.py` already does some testing for various OS for copying the hosts file into place.  There's no reason why that couldn't be done earlier, and poke-in OS-specific default hosts automatically.  For the public amalgamated hosts file, we could include 'em all.

**Comment by matkoniecz** [comment]
Or maybe on the first run (though trying to detect it reliably is going to be problematic) just pull copy existing hosts file to `myhosts`?

**Comment by StevenBlack** [comment]
This is fixed in commit ab12924 on Feb 19 2016.

Closing.
